server/cache.go

```diff
@@ -225,6 +225,13 @@ func (r *regionsInfo) randFollowerRegion(storeID uint64) *regionInfo {
 	return randRegion(r.followers[storeID])
 }
 
+func (r *regionsInfo) randomRegion() *regionInfo {
+	for _, region := range r.regions {
```

Comment:
Then we have to use another `[]uint64` to record all region ids and sync it with regions map. I think we can rely on the fact that maps are iterated from random position.